LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/ClientCertificateValidatorService.cs

```diff
@@ -43,13 +43,18 @@ public async Task<bool> ValidateAsync(X509Certificate2 certificate, X509Chain? c
 
             using var scope = this.logger.BeginEuiScope(stationEui);
 
-            // Logging any chain related issue, but not failing on it.
-            foreach (var status in chain.ChainStatus)
+            // Logging any chain related issue that is causing verification to fail
+            if (chain.ChainStatus.Any())
```

Comment:
Thisd part is not clear to me, I understood it reprensents the certificate chain of the submitted certificate?  Why would the presence of chain status be a sign of error, couldn't that be also correct case? https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.x509certificates.x509chainstatus?view=net-6.0

Reply:
Would be nice if we could have a chat

Reply:
After the offline discussion, I added a condition on the check that is verifying if there is any chainstatus flag that is not corresponding to the "NoError" one.